### PR TITLE
[WIP] feat: get clam price from dyst

### PR DESCRIPTION
**Pull request by pyc92**

### CODE REVIEW

**Review by abtheo — APPROVED** [review]
Awesome, code is much cleaner! New functionality looks like it works but it's always helpful to have a development subgraph synced for testing before we deploy to main. I'll start syncing it on my dev graph here - https://thegraph.com/hosted-service/subgraph/abtheo/otter-development?pending

**Review by abtheo — CHANGES_REQUESTED** [review]
Ah, unfortunately looks like it fails to sync - 
image

```
Subgraph error 1/3, code: SubgraphSyncingFailure, error: Mapping aborted at ~lib/@graphprotocol/graph-ts/chain/ethereum.ts, line 434, column 7, with message: Call reverted, probably because an `assert` or `require` in the contract failed, consider using `try_decimals` to handle this in the mapping. wasm backtrace: 0: 0x6247 - <unknown>!~lib/@graphprotocol/graph-ts/chain/ethereum/ethereum.SmartContract#call 1: 0x66df - <unknown>!src/utils/Price/findTokenPrice 2: 0x6b0b - <unknown>!src/utils/Price/getClamUsdRate 3: 0x6fa5 - <unknown>!src/utils/Price/getPairUSD 4: 0x77ae - <unknown>!src/utils/ProtocolMetrics/setTreasuryAssetMarketValues 5: 0x9dff - <unknown>!src/utils/ProtocolMetrics/updateProtocolMetrics 6: 0xa7dc - <unknown>!src/StakedOtterClamERC20V2/handleTransfer in handler `handleTransfer` at block #21593528 (9caf2ced725c8946caf9f28c81448fb4f6322981ee37110d7b67082a84ac34bf), block_hash:
```

**Comment by pyc92** [comment]
@abtheo  this one is still work in progress, I will fix it after we release staking v2.

File: src/utils/Price.ts [hunk 1]
```diff
@@ -37,128 +49,91 @@ export function getwMaticUsdRate(): BigDecimal {
...
+export function getClamUsdRate(block: BigInt): BigDecimal {
+  let rate: BigDecimal
 
-  if (clam.equals(BigDecimal.zero()) || mai.equals(BigDecimal.zero())) {
-    log.debug('getCLAMUSDRate div {}', [clam.toString()])
-    return BigDecimal.zero()
+  if (block.gt(BigInt.fromI32(DYST_POOL_TRANSITION_BLOCK))) {
+    rate = findTokenPrice(dyst, CLAM_ERC20, USDPLUS_ERC20)
+  } else {
+    rate = findTokenPrice(quickSwap, CLAM_ERC20, USDPLUS_ERC20)
```
Comment (abtheo) [inline src/utils/Price.ts:116]: Ah, I think I found the issue 😁

Before we moved to Dystopia, our liquidity pool was CLAM-MAI, not CLAM-USD+. 

`rate = findTokenPrice(quickSwap, CLAM_ERC20, USDPLUS_ERC20)` ==> `rate = findTokenPrice(quickSwap, CLAM_ERC20, MAI_ERC20)`

I am currently working on a large update for the Graph, and I will merge this PR into my branch. I'll fix this issue as well. 👍

Comment (otterclamking) [reply src/utils/Price.ts:116]: Great! Thank you @abtheo

**Pull request closed without merging**